core/src/main/java/org/apache/iceberg/util/StructLikeMap.java

```diff
@@ -104,7 +104,7 @@ public void clear() {
   }
 
   @Override
-  public Set<StructLike> keySet() {
```

Comment:
I'm not sure we can change these since we'll be breaking an api that is more or less public .... I need to think on this and the StructLikeSet methods ... 

I'm pretty sure we don't want to break the api, but we also don't want to encourage folks using raw Set's and Maps of StructLike but in this case they would be safe since we know they came from StructLikeMap and StructLikeSet ....

Reply:
Sometimes can use the original, sometimes can't，Then rule `<property name="format" value="^Map&lt;StructLike, .*&gt;"/>` may not be able to cover the whole.

Reply:
Yes, but in that case we put an exception for this signature. All checkstyle rules allow for overrides for particular lines, we don't need to have the check style rule be perfect.

Reply:
ok, I think we revert that first, If has problem can use SuppressWarnings annotation.

Reply:
This is the change I think we need to revert, I get worried about changing the public api of this class. I think just suppressing the rule for this usage is fine

Reply:
ok